module-4/nmap/nmap.py

```diff
@@ -0,0 +1,80 @@
+import socket
+import argparse
+import concurrent.futures
+
+# Common services for the first 100 ports
+COMMON_PORTS = {
+    20: "FTP data",
+    21: "FTP control",
+    22: "SSH",
+    23: "Telnet",
+    25: "SMTP",
+    53: "DNS",
+    80: "HTTP",
+    110: "POP3",
+    143: "IMAP",
+    443: "HTTPS",
+    993: "IMAPS",
+    995: "POP3S",
+    # ... continue up to port 100 as needed
+}
+
+def resolve_host(host):
+    try:
+        ip = socket.gethostbyname(host)
+        print(f"[INFO] IP address for {host} is {ip}")
+        return ip
+    except socket.gaierror:
+        print(f"[ERROR] Could not resolve host {host}.")
+        return None
+
+def scan_port(ip, port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        s.settimeout(1)
+        result = s.connect_ex((ip, port))
+        if result == 0:
+            service_name = COMMON_PORTS.get(port, "Unknown Service") if port <= 100 else "Unknown"
```

Comment:
not sure if this is a follow-up error with the 100. In my opinion this should be 1000 here, too. Otherwise it does not make sense to list codes until 1000 in your `COMMON_PORTS` dictionary.